ml_tools/eolearn/tests/test_sampling.py

```diff
@@ -92,82 +93,107 @@ def test_sample_by_values_errors(image: np.ndarray, n_samples: Dict[int, int]) -
-    eop.mask_timeless["raster"] = small_image.reshape(small_image.shape + (1,))
+    eopatch = EOPatch()
+    eopatch.data["bands"] = np.arange(t * h * w * d).reshape(t, h, w, d)
+    eopatch.mask_timeless["raster"] = small_image.reshape(small_image.shape + (1,))
+    return eopatch
 
-    task = BlockSamplingTask(
+
+SAMPLING_MASK = FeatureType.MASK_TIMELESS, "sampling_mask"
+
+
+@pytest.fixture(name="block_task")
+def block_task_fixture(request) -> EOTask:
+    return BlockSamplingTask(
         [(FeatureType.DATA, "bands", "SAMPLED_DATA"), (FeatureType.MASK_TIMELESS, "raster", "SAMPLED_LABELS")],
-        amount=amount,
-        mask_of_samples=(FeatureType.MASK_TIMELESS, "sampling_mask"),
+        amount=request.param,
+        mask_of_samples=SAMPLING_MASK,
     )
 
-    task.execute(eop, seed=seed)
+
+@pytest.mark.parametrize("seed", range(5))
+@pytest.mark.parametrize("block_task", [100, 5231, 0.4, 0], indirect=True)
+def test_object_sampling_task_mask(
+    eopatch: EOPatch, small_image: np.ndarray, seed: int, block_task: BlockSamplingTask
+) -> None:
+    t, h, w, d = 10, *small_image.shape, 5
+    amount = block_task.amount
+
+    block_task.execute(eopatch, seed=seed)
     expected_amount = amount if isinstance(amount, int) else round(np.prod(small_image.shape) * amount)
 
-    # assert features, labels and sampled rows and cols are added to eopatch
-    assert "SAMPLED_LABELS" in eop.mask_timeless, "Labels not added to eopatch"
-    assert "SAMPLED_DATA" in eop.data, "Features not added to eopatch"
-    assert "sampling_mask" in eop.mask_timeless, "Mask of sampling not generated"
     # check validity of sampling
-    assert eop.data["SAMPLED_DATA"].shape == (t, expected_amount, 1, d), "Incorrect features size"
-    assert eop.mask_timeless["SAMPLED_LABELS"].shape == (expected_amount, 1, 1), "Incorrect number of samples"
-    assert eop.mask_timeless["sampling_mask"].shape == (h, w, 1), "Sampling mask of incorrect size"
+    assert eopatch.data["SAMPLED_DATA"].shape == (t, expected_amount, 1, d)
+    assert eopatch.mask_timeless["SAMPLED_LABELS"].shape == (expected_amount, 1, 1)
+    assert eopatch.mask_timeless["sampling_mask"].shape == (h, w, 1)
 
-    sampled_uniques, sampled_counts = np.unique(eop.data["SAMPLED_DATA"], return_counts=True)
-    masked = eop.mask_timeless["sampling_mask"].squeeze(axis=2) == 1
-    masked_uniques, masked_counts = np.unique(eop.data["bands"][:, masked, :], return_counts=True)
-    assert_array_equal(sampled_uniques, masked_uniques, err_msg="Sampling mask not correctly describing sampled points")
-    assert_array_equal(sampled_counts, masked_counts, err_msg="Sampling mask not correctly describing sampled points")
 
+@pytest.mark.parametrize("seed", range(5))
+@pytest.mark.parametrize("block_task", [100, 5231, 0.4, 0], indirect=True)
+def test_object_sampling_task(eopatch: EOPatch, seed: int, block_task: BlockSamplingTask) -> None:
+    block_task.execute(eopatch, seed=seed)
```

Comment:
i would include that in the previous test

Reply:
done